lib/galaxy/tools/wrappers.py

```
@@ -80,14 +80,12 @@ def __init__(self, input, value, other_values={}):
         self._other_values = other_values
 
     def __eq__(self, other):
-        if isinstance(other, string_types):
+        if isinstance(other, string_types) or (isinstance(other, InputValueWrapper) and other.input.type == 'text'):
             return str(self) == other
-        elif isinstance(other, int):
-            return int(self) == other
-        elif isinstance(other, float):
+        elif isinstance(other, (int, float)) or (isinstance(other, InputValueWrapper) and other.input.type in ('float', 'integer')):
             return float(self) == other
         else:
-            return super(InputValueWrapper, self) == other
+            return self.value == other
```

Not sure I'd go that far.

---

We could instead raise an exception here, but this should work for comparing color parameters for instance.

---

I'd prefer to cross that bridge when there is a reported issue/use case, especially since this PR is targeting 19.05.

---

I agree with the sentiment, but what's your suggestion ? This doesn't work at all right now (none of the parent classes define __gt__)